environs/filestorage/filestorage.go

```diff
@@ -159,8 +159,9 @@ func (f *fileStorageWriter) Put(name string, r io.Reader, length int64) error {
 	if err != nil {
 		return err
 	}
+	defer func() { _ = file.Close() }()
+
```

Comment:
Is there an advantage of putting the statement into a thunk? At first I thought it was for handling the error within defer but you seem to be ignoring it anyways.

I see in other spots you used closures like this, which I'm thinking for having an order in which they are executed, right?